### PR TITLE
Add Slider component

### LINKED ISSUE #160: Add Slider component
Consider adding the mdc-slider https://github.com/material-components/material-components-web/tree/master/packages/mdc-slider

### PR DESCRIPTION
**Pull request by MobiusHorizons**
Fixes #160

### CODE REVIEW

File: Slider/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import { MDCSlider } from "@material/slider";
+
+/**
+ * @prop disabled = false
+ */
+export default class Slider extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "slider";
+    this._mdcProps = ["disabled", "min", "max", "value", "step"];
```
Comment (prateekbh) [inline Slider/index.js]: So the way _mdcProps works is:
if user gives the value as true/false then these will be converted to specific `mdc_componentname--classname`

https://github.com/prateekbh/preact-material-components/blob/master/MaterialComponent.js#L35-L37

So I dont this these are needed here

File: Slider/Slider.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { MDCSlider } from "@material/slider";
+
+/**
+ * @prop disabled = false
+ */
+export default class Slider extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "slider";
+    this._mdcProps = ["disabled", "min", "max", "value", "step"];
```
Comment (prateekbh) [inline Slider/Slider.jsx]: So the way _mdcProps works is:
if user gives the value as true/false then these will be converted to specific mdc_componentname--classname

https://github.com/prateekbh/preact-material-components/blob/master/MaterialComponent.js#L35-L37

So I dont this these are needed here

File: Slider/Slider.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    if (step) {
+      dynamicProps["data-step"] = step;
+    }
+
+    return (
+      <div
+        class={className + " mdc-slider"}
+        tabindex="0"
```
Comment (prateekbh) [inline Slider/Slider.jsx]: can we derive 0 as override-able prop?

File: Slider/Slider.jsx (hunk 2 above)
Comment (MobiusHorizons) [inline Slider/Slider.jsx]: oh ok, i'll remove them.

File: Slider/Slider.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      dynamicProps["aria-disabled"] = true;
+    }
+
+    if (step) {
+      dynamicProps["data-step"] = step;
+    }
+
+    return (
+      <div
+        class={className + " mdc-slider"}
```
Comment (MobiusHorizons) [inline Slider/Slider.jsx]: got it.

File: Slider/Slider.jsx (hunk 3 above)
Comment (MobiusHorizons) [inline Slider/Slider.jsx]: I'll add it in.

File: Slider/Slider.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      dynamicProps["aria-disabled"] = true;
+    }
+
+    if (step) {
+      dynamicProps["data-step"] = step;
+    }
+
+    return (
+      <div
+        class={className + ""}
```
Comment (prateekbh) [inline Slider/Slider.jsx]: I dont think a separate className is required at all ... dynamicProps will put it here

File: Slider/Slider.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        tabindex={tabindex}
+        role="slider"
+        aria-valuemin={min}
+        aria-valuemax={max}
+        aria-valuenow={value}
+        aria-label="Select Value"
+        {...dynamicProps}
+      >
```
Comment (prateekbh) [inline Slider/Slider.jsx]: also add `...props` here, so that anything extra a User wanna add, they can

Comment (MobiusHorizons) [inline Slider/Slider.jsx]: ok, i was sort of going back and forth on that one. If you are cool with that, i'll throw it in

File: Slider/Slider.jsx (hunk 5 above)
Comment (MobiusHorizons) [inline Slider/Slider.jsx]: oh, i misunderstood this one, let me clean that up.

**Comment by MobiusHorizons** [comment]
ok, this should be good now. I also rebased and squashed so it's just one commit in the history.

**Comment by prateekbh** [comment]
hey, I am publishing this now, do you want to add this to `gh-pages` branch or shall I go ahead and do it?

**Comment by MobiusHorizons** [comment]
I can probably do it this weekend, but if you get to it first, I certainly wouldn't mind. 
Thank you for the great library!!

**Comment by prateekbh** [comment]
i've update the docs as well :)

**Pull request merged**